File: surf-api-core/surf-api-core-api/src/main/java/dev/slne/surf/surfapi/core/api/command/exception/WrapperCommandExceptionComponent.java
package dev.slne.surf.surfapi.core.api.command.exception;

import com.mojang.brigadier.exceptions.CommandSyntaxException;
import com.mojang.brigadier.exceptions.SimpleCommandExceptionType;
import dev.jorel.commandapi.exceptions.WrapperCommandSyntaxException;
import dev.slne.surf.surfapi.core.api.messages.ComponentMessage;
import java.io.Serial;
import net.kyori.adventure.text.Component;

public class WrapperCommandExceptionComponent extends WrapperCommandSyntaxException {

  @Serial
  private static final long serialVersionUID = -5277462249842682916L;

  public WrapperCommandExceptionComponent(Component errorMessage) {
    super(new CommandSyntaxException(
        new SimpleCommandExceptionType(new ComponentMessage(errorMessage)),
        new ComponentMessage(errorMessage)));
  }

  @Override
  public String getMessage() {
    return getRawMessage().getString();
  }
}

File: surf-api-core/surf-api-core-api/src/main/kotlin/dev/slne/surf/surfapi/core/api/command/exception/WrapperCommandExceptionComponent.kt
package dev.slne.surf.surfapi.core.api.command.exception

import com.mojang.brigadier.exceptions.CommandSyntaxException
import com.mojang.brigadier.exceptions.SimpleCommandExceptionType
import dev.jorel.commandapi.exceptions.WrapperCommandSyntaxException
import dev.slne.surf.surfapi.core.api.messages.ComponentMessage
import net.kyori.adventure.text.Component
import java.io.Serial

/**
 * A wrapper for [CommandSyntaxException] to integrate custom error messages
 * as [Component] objects within the Surf API.
 *
 * @param errorMessage The error message to associate with the exception.
 */
class WrapperCommandExceptionComponent(errorMessage: Component) : WrapperCommandSyntaxException(
    CommandSyntaxException(
        SimpleCommandExceptionType(
            ComponentMessage(errorMessage)
        ), ComponentMessage(errorMessage)
    )
) {
    override val message: String?
        get() = rawMessage.string

    companion object {
        @Serial
        private val serialVersionUID = -5277462249842682916L
    }
}
